sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/UserDefinedFunctionErrors.scala

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.catalog
+
+import org.apache.spark.sql.errors.QueryErrorsBase
+
+/**
+ * Errors during registering and executing [[UserDefinedFunction]]s.
+ */
+object UserDefinedFunctionErrors extends QueryErrorsBase {
+  def unsupportedUserDefinedFunction(language: RoutineLanguage): Throwable = {
+    unsupportedUserDefinedFunction(language.name)
+  }
+
+  def unsupportedUserDefinedFunction(language: String): Throwable = {
+    new IllegalArgumentException(s"Unsupported user defined function type: $language")
```

Comment:
shall we add an error class?

Reply:
This should be an internal error since we blocked languages other than SQL in the parser.